Lib/test/test_subprocess.py

```diff
@@ -1214,7 +1214,9 @@ def test_communicate_errors(self):
 
     def test_no_leaking(self):
         # Make sure we leak no resources
-        if not mswindows:
+        if sys.platform.startswith("netbsd"):
```

Comment:
rather than more specialization for what is probably a kernel config... I think catching ValueError within the loop down where OSError is caught and treating it the same would make more sense.

Reply:
issue isn't raising here;
```
                try:
                    tmpfile = os.path.join(tmpdir, os_helper.TESTFN)
                    handles.append(os.open(tmpfile, os.O_WRONLY|os.O_CREAT))
                except OSError as e:
                    if e.errno != errno.EMFILE:
                        raise
                    break

Reply:
I'm sorry, I don't know exactly what you mean. The error is not being raised in this context.
```python
try:
    tmpfile = os.path.join(tmpdir, os_helper.TESTFN)
    handles.append(os.open(tmpfile, os.O_WRONLY|os.O_CREAT))
except OSError as e:
    if e.errno != errno.EMFILE:
        raise
    break
```

Is this what you mean?
```python
try:
    p = subprocess.Popen([sys.executable, "-c",
                          "import sys;"
                          "sys.stdout.write(sys.stdin.read())"],
                         stdin=subprocess.PIPE,
                         stdout=subprocess.PIPE,
                         stderr=subprocess.PIPE)
    data = p.communicate(b"lime")[0]
    self.assertEqual(data, b"lime")
except Exception as err:
    self.skipTest("failed to reach the file descriptor limit "
         "(tried %d)" % max_handles)
```